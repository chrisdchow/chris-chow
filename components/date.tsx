import { parseISO, format } from 'date-fns';
import { FunctionComponent } from 'react';

type DateProps = {
  dateString: string;
};

export const Date: FunctionComponent<DateProps> = ({ dateString }) => {
  const date = parseISO(dateString);
  return <time dateTime={dateString}>{format(date, 'LLLL d, yyyy')}</time>;
};
