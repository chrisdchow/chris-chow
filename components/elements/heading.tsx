import { FunctionComponent } from 'react';

type HeadingProps = Record<string, unknown>;

/**
 * https://tailwindcss.com/docs/preflight
 * Headings are unstyled (normalized)
 */

const baseHeadingStyle = 'font-bold leading-loose my-4 mx-0';

export const H1: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h1 className={`${baseHeadingStyle} text-3xl`}>{children}</h1>;
};

export const H2: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h2 className={`${baseHeadingStyle} text-2xl`}>{children}</h2>;
};

export const H3: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h3 className={`${baseHeadingStyle} text-xl`}>{children}</h3>;
};

export const H4: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h4 className={`${baseHeadingStyle} text-l`}>{children}</h4>;
};
