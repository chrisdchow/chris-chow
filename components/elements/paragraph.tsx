import { FunctionComponent } from 'react';

type ParagraphProps = Record<string, unknown>;

/**
 * https://tailwindcss.com/docs/preflight
 * Paragraph tags are unstyled (normalized)
 */

export const P: FunctionComponent<ParagraphProps> = ({ children }) => {
  return <p className='my-2 whitespace-pre-wrap'>{children}</p>;
};
