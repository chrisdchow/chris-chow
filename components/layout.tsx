import { FunctionComponent } from 'react';
import Head from 'next/head';
import Link from 'next/link';
import { H1, H2 } from '@components/elements/heading';

const name = 'Chris Chow';
export const siteTitle = name;

type LayoutProps = {
  home?: boolean;
};

export const Layout: FunctionComponent<LayoutProps> = ({ home, children }): JSX.Element => {
  const homeHeader = (
    <>
      <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
      <H1>{name}</H1>
    </>
  );

  const navigateHomeHeader = (
    <>
      <Link href='/'>
        <a>
          <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
        </a>
      </Link>
      <H2>
        <Link href='/'>
          <a>{name}</a>
        </Link>
      </H2>
    </>
  );

  return (
    <div className='max-w-xl py-0 px-4 mx-auto my-12'>
      <Head>
        <link rel='icon' href='/favicon.ico' />
        <meta name='description' content='Chris Chow' />
        <meta property='og:image' content={`https://og-image.now.sh/${encodeURI(siteTitle)}.png`} />
        <meta name='og:title' content={siteTitle} />
        <meta name='twitter:card' content='summary_large_image' />
      </Head>
      <header className='flex flex-col items-center'>{home ? homeHeader : navigateHomeHeader}</header>
      <main>{children}</main>
      {!home && (
        <div className='mt-12 mx-0 mb-0'>
          <Link href='/'>
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  );
};
