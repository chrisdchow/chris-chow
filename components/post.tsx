import { FunctionComponent } from 'react';
import parse, { domToReact } from 'html-react-parser';
import Head from 'next/head';
import { Layout } from '@components/layout';
import { Date } from '@components/elements/date';
import { H2 } from '@components/elements/heading';
import { P } from '@components/elements/paragraph';

type PostProps = {
  postData: {
    id: string;
    date: any;
    title: string;
    contentHtml: string;
  };
};

export const Post: FunctionComponent<PostProps> = ({ postData }) => {
  const content = parse(postData.contentHtml, {
    replace: function ({ type, name, children }) {
      if (type === 'tag' && name === 'p') {
        return <P>{domToReact(children)}</P>;
      }
    },
  });

  return (
    <Layout>
      <Head>
        <title>{postData.title}</title>
      </Head>
      <article>
        <H2>{postData.title}</H2>
        <div className='text-gray-400'>
          <Date dateString={postData.date} />
        </div>
        {content}
      </article>
    </Layout>
  );
};
