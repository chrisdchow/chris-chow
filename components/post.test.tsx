import React from 'react';
import { render } from '@test/test-utils';
import { Post } from './post';

describe('Post compnent', () => {
  it('matches snapshot', () => {
    const postData = {
      id: 'testPost',
      date: '2020-01-02',
      title: 'Test Post',
      contentHtml: '',
    };
    const { asFragment } = render(<Post postData={postData} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
