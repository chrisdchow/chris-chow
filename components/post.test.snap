// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post compnent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <a
        href="/"
      >
        <img
          alt="Chris Chow"
          class="headerImage borderCircle"
          src="/images/profile.jpg"
        />
      </a>
      <h2
        class="headingLg"
      >
        <a
          class="colorInherit"
          href="/"
        >
          Chris Chow
        </a>
      </h2>
    </header>
    <main>
      <article>
        <h1
          class="headingXl"
        >
          Test Post
        </h1>
        <div
          class="lightText"
        >
          <time
            datetime="2020-01-02"
          >
            January 2, 2020
          </time>
        </div>
        <div />
      </article>
      <div />
    </main>
    <div
      class="backToHome"
    >
      <a
        href="/"
      >
        ‚Üê Back to home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
